src/dbnode/persist/types.go

```diff
@@ -180,20 +181,30 @@ type CommitLogFile struct {
 	Index    int64
 }
 
-// IndexFn is a function that persists a m3ninx MutableSegment.
-type IndexFn func(segment.Builder) error
+// IndexFlushFn is a function that persists a m3ninx MutableSegment.
+type IndexFlushFn func(segment.Builder) error
+
+// IndexSnapshotFn is a function that persists fst SegmentData.
+type IndexSnapshotFn func(fst.SegmentData) error
 
 // IndexCloser is a function that performs cleanup after persisting the index data
 // block for a (namespace, blockStart) combination and returns the corresponding
 // immutable Segment.
 type IndexCloser func() ([]segment.Segment, error)
 
-// PreparedIndexPersist is an object that wraps holds a persist function and a closer.
-type PreparedIndexPersist struct {
-	Persist IndexFn
+// PreparedIndexFlushPersist is an object that wraps holds a persist function and a closer.
+type PreparedIndexFlushPersist struct {
+	Persist IndexFlushFn
 	Close   IndexCloser
 }
 
+// PreparedIndexSnapshotPersist is an object that wraps holds a persist function and a closer.
```

Comment:
nit: comment is a duplicate of the one above. Probably worth pointing out the difference in one of the two